src/python/pants/backend/python/tasks/python_execution_task_base.py

```diff
@@ -76,6 +111,10 @@ def create_pex(self, pex_info=None):
           pexes = [extra_requirements_pex] + pexes
         constraints = {constraint for rt in relevant_targets if is_python_target(rt)
                        for constraint in rt.compatibility}
-        self.merge_pexes(path, pex_info, interpreter, pexes, constraints)
+
+        with self.merged_pex(path, pex_info, interpreter, pexes, constraints) as builder:
+          for extra_file in self.extra_files():
+            extra_file.add_to(builder)
```

Comment:
IIRC, pex already handles exploding for duped files, so it should be safe to unconditionally add the `pytest.ini`.

Reply:
I'm not following. I do no deduping here.

Reply:
Sorry, was convincing myself (out loud) that a collision would be detected here.

Reply:
Nope, pex silently overwrites for any given path+type if path and type match (where type is source vs resource for example). The specific case you mention will never have a conflict since the pytest-prep pex is approximately deps-only (py.test), but now adds 2 loose-source files, a `pytest.ini` and the Pants coverage plugin. The pytest-prep pex has the structure:
```
pytest-prep/
  .deps/
  .bootstrap/
  PEX-INFO: merge [
    1. user test code graph reqs,
    2. user test code graph sources,
    3. py.test req
  ] 
  pytest.ini
  pants/
    __init__.py (ns package)
    backend/
      __init__.py (ns package)
      python/
        __init__.py (ns package)
        tasks/
          __init__.py (ns package)
          coverage/
            __init__.py
            plugin.py
  __main__.py
```
As a result, the pytest-prep `pytest.ini` file trumps being 0th on the merge path since its embedded in the host of the merge.